project_euler/problem_11.py

```diff
@@ -0,0 +1,99 @@
+        [88, 36, 68, 87, 57, 62, 20, 72, 03, 46, 33, 67, 46, 55, 12, 32, 63, 93, 53, 69],
+        [4, 42, 16, 73, 38, 25, 39, 11, 24, 94, 72, 18, 8, 46, 29, 32, 40, 62, 76, 36],
+        [20, 69, 36, 41, 72, 30, 23, 88, 34, 62, 99, 69, 82, 67, 59, 85, 74, 04, 36, 16],
+        [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 05, 54],
+        [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]]
+
+
+def get_right_seq_multiplication(i, j):
+    try:
+        return grid[i][j] * grid[i][j + 1] * grid[i][j + 2] * grid[i][j + 3]
+    except IndexError:
+        return 0
+
+
+def get_right_bottom_diagonal_seq_multiplication(i, j):
+    try:
+        return grid[i][j] * grid[i + 1][j + 1] * grid[i + 2][j + 2] * grid[i + 3][j + 3]
+    except IndexError:
+        return 0
+
+
+def get_bottom_seq_multiplication(i, j):
+    try:
+        return grid[i][j] * grid[i + 1][j] * grid[i + 2][j] * grid[i + 3][j]
+    except IndexError:
+        return 0
+
+
+def get_left_bottom_diagonal_seq_multiplication(i, j):
+    try:
+        return grid[i][j] * grid[i + 1][j - 1] * grid[i + 2][j - 2] * grid[i + 3][j - 3]
+    except IndexError:
+        return 0
+
+
+def get_left_seq_multiplication(i, j):
+    try:
+        return grid[i][j] * grid[i][j - 1] * grid[i][j - 2] * grid[i][j - 3]
+    except IndexError:
+        return 0
+
+
+def get_left_top_diagonal_seq_multiplication(i, j):
+    try:
+        return grid[i][j] * grid[i - 1][j - 1] * grid[i - 2][j - 2] * grid[i - 3][j - 3]
+    except IndexError:
+        return 0
+
+
+def get_top_seq_multiplication(i, j):
+    try:
+        return grid[i][j] * grid[i - 1][j] * grid[i - 2][j] * grid[i - 3][j]
+    except IndexError:
+        return 0
+
+
+def get_right_top_diagonal_seq_multiplication(i, j):
+    try:
+        return grid[i][j] * grid[i - 1][j + 1] * grid[i - 2][j + 2] * grid[i - 3][j + 3]
+    except IndexError:
+        return 0
+
+
+def main():
+    multiplication_products = []
+    for i in range(20):
+        for j in range(20):
+            pivot = grid[i][j]
+            multiplication_products.append(get_right_seq_multiplication(i, j))
+            multiplication_products.append(get_right_bottom_diagonal_seq_multiplication(i, j))
+            multiplication_products.append(get_bottom_seq_multiplication(i, j))
+            multiplication_products.append(get_left_bottom_diagonal_seq_multiplication(i, j))
+            multiplication_products.append(get_left_seq_multiplication(i, j))
+            multiplication_products.append(get_left_top_diagonal_seq_multiplication(i, j))
+            multiplication_products.append(get_top_seq_multiplication(i, j))
+            multiplication_products.append(get_right_top_diagonal_seq_multiplication(i, j))
+            multiplication_products.append(get_left_seq_multiplication(i, j))
+    print "Max multiplication product is: %s" % max(multiplication_products)
+
+
```

Comment:
There're some `pylint` warnings to be fixed. (It's a good idea to have this check built-in to the editor).